### PR TITLE
Add technical lead role description

### PR DESCRIPTION
**Pull request by saschagrunert**
This commit adds a role description for technical leads in the
Kubernetes community.

**Which issue(s) this PR fixes**:

None

/cc @mbbroberg @justaugustus @alejandrox1 @LappleApple @parispittman

### CODE REVIEW

File: contributors/chairs-and-techleads/README.md [hunk 1]
```diff
@@ -1,38 +1,112 @@
...
+- **Architecture** – Wide range understanding of the SIGs work in correlation to
+  the whole community. Establishing future plans by continuously working on the
+  technical vision of the SIG.
+
+If you are interested in becoming a Technical Lead, speak with the [appropriate
+SIG Chairs](https://github.com/kubernetes/community/blob/master/sig-list.md).
+SIG Chairs will either let you know what skills you need to develop to be
+eligible or help you through the process of becoming a Technical Lead.
+
+### Resources for community group leads
```
Comment (saschagrunert) [inline contributors/chairs-and-techleads/README.md:76]: Not sure what we want to do with this section.

**Comment by nikhita** [comment]
Going to add a hold since this will need steering review
cc @kubernetes/steering-committee 

/hold
/committee steering

**Comment by mbbroberg** [comment]
Hey @saschagrunert -- I was thinking this is the right folder, but would be better as a discrete document since it (rather well!) explains the TL role only. This README needs more improvements and would benefit from noting its contents, including this new file. 

Could you move this info to a `teachnical-lead.md` file, and we can revise the README in another PR?

**Comment by saschagrunert** [comment]
> Could you move this info to a `teachnical-lead.md` file, and we can revise the README in another PR?

Sure, changed as suggested :+1:

**Comment by mbbroberg** [comment]
@saschagrunert from a readership point of view, this is clear, concise, and helps a reader know how to be and become a TL. 

I'm following @nikhita's lead that steering needs to give a thumbs up or down on it. CC @parispittman so you know I helped shepherd from the Slack chat you tagged our team in. 

/approve 
/lgtm

**Review by mbbroberg — APPROVED** [review]

**Comment by saschagrunert** [comment]
Hey, any update about this one? Can I do anything to get this merged? :innocent:

File: contributors/chairs-and-techleads/teachnical-lead.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+- **Development** – Knowledge about the code and the quality standards within
+  the community. Being able to help the team to solve technical obstacles
+  without having a need of being the expert for everything.
+- **Architecture** – Wide range understanding of the SIGs work in correlation to
+  the whole community. Establishing future plans by continuously working on the
+  technical vision of the SIG.
+
+If you are interested in becoming a Technical Lead, speak with the [appropriate
+SIG Chairs](https://github.com/kubernetes/community/blob/master/sig-list.md).
+SIG Chairs will either let you know what skills you need to develop to be
```
Comment (dims) [inline contributors/chairs-and-techleads/teachnical-lead.md:73]: I think we can drop this sentence. 73-74.

Comment (saschagrunert) [reply contributors/chairs-and-techleads/teachnical-lead.md:73]: Yep, removed that one :+1:

File: contributors/chairs-and-techleads/teachnical-lead.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  change, then it is in the responsibility of the Technical Leads to drive
+  towards a feasible solution for the whole community.
+
+Technical leadership within a SIG should focus on solving the “How” questions
+rather than people and team growth related ones (the latter responsibilities are
+those of the SIG Chairs). Mentoring team members around solving technical tasks
+also falls also into the responsibility of the Technical Leads. This includes
+proper onboarding (or delegation of it) of new team members with respect to
+technical workflows within the SIG. Technical Leads also help mediating
+technical debates within and in correlation with other SIGs. They actively
```
[the remark this answers is not in the document]
Comment (nikhita) [reply contributors/chairs-and-techleads/teachnical-lead.md:56]: re: debate - to make it more explicit for someone new reading this maybe we could also include something like "the ability to ask the right questions and say no"?

**Review by nikhita — COMMENTED** [review]
@saschagrunert This is great, thank you so much for writing this down!!

Left some suggestions but I'm also ok with moving them to a follow-up if that's easier. :)

File: contributors/chairs-and-techleads/teachnical-lead.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (saschagrunert) [reply contributors/chairs-and-techleads/teachnical-lead.md]: Yep, added the points to the sentence. :+1:

File: contributors/chairs-and-techleads/teachnical-lead.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+- Take changing environments into account to adapt the technical vision if
+  needed. For example, if interdependent SIGs have a requirement for a technical
+  change, then it is in the responsibility of the Technical Leads to drive
+  towards a feasible solution for the whole community.
+
+Technical leadership within a SIG should focus on solving the “How” questions
+rather than people and team growth related ones (the latter responsibilities are
+those of the SIG Chairs). Mentoring team members around solving technical tasks
+also falls also into the responsibility of the Technical Leads. This includes
+proper onboarding (or delegation of it) of new team members with respect to
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply contributors/chairs-and-techleads/teachnical-lead.md:54]: Yes, added this information as suggested. :+1:

File: contributors/chairs-and-techleads/teachnical-lead.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+### Abstract
+
+The Technical Lead role in Kubernetes is an optional role that each SIG can
+choose to implement as part of its [governance
+model](https://github.com/kubernetes/community/blob/3b283852567872e38f0b97beb20d918667882754/committee-steering/governance/sig-governance.md#roles).
+This means SIGs can decide on their own if they want to add Technical Leads to
+their charter or not. Depending on the overall size of the SIG, around two to
+three people can be chosen by the SIG Chairs to support the technical aspects of
+the group. To be able to fulfill their role, a Technical Lead should have the
+same set of permissions as a Chair.
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply contributors/chairs-and-techleads/teachnical-lead.md:21]: I think every SIG chairs have their own set of necessary permissions, right? Not sure if we more it more explicit.

File: contributors/chairs-and-techleads/teachnical-lead.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+Generally speaking, Technical Leads are responsible for leading the SIG in
+correspondence with its technical alignment. This alignment includes both
+internal to the SIG and, more significantly, external to the entire Kubernetes
+project. Before being able to align any technical direction, it is necessary to
+establish a technical vision within the SIG. The technical vision should be
+continuously updated and turned into reality, whereas larger features of the
+vision can be outlined in dedicated roadmaps. This means Technical Leads have to
+be involved in the source code base of the SIG to be able to take the right
+decisions, identify risk and maintain a high level of trust with other members
+of the SIG.
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply contributors/chairs-and-techleads/teachnical-lead.md:32]: Hm, I can follow-up on that. :)

**Comment by nikhita** [comment]
Lgtm, we can always iterate if needed :)

/cc @kubernetes/steering-committee 
PTAL

File: contributors/chairs-and-techleads/teachnical-lead.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+Target of this document is to define and outline the Technical Lead role within
+the Kubernetes community. The document can be used as guidance for Special
+Interest Groups (SIGs) to onboard new Technical Leads as well as clarifying the
+expectations associated with this role.
+
+### Abstract
+
+The Technical Lead role in Kubernetes is an optional role that each SIG can
+choose to implement as part of its [governance
+model](https://github.com/kubernetes/community/blob/3b283852567872e38f0b97beb20d918667882754/committee-steering/governance/sig-governance.md#roles).
```
Comment (liggitt) [inline contributors/chairs-and-techleads/teachnical-lead.md:16]: suggest using one of these URLs that will link to the latest content:

* absolute via redirect: http://git.k8s.io/community/committee-steering/governance/sig-governance.md#roles 
* relative: /committee-steering/governance/sig-governance.md#roles

File: contributors/chairs-and-techleads/teachnical-lead.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  the whole Kubernetes organization. One example would be using Prow for
+  Continuous Integration (CI) practices.
+- Continuously evaluate technical challenges within the SIG and work towards
+  removing them as part of the vision.
+- Take changing environments into account to adapt the technical vision if
+  needed. For example, if interdependent SIGs have a requirement for a technical
+  change, then it is in the responsibility of the Technical Leads to drive
+  towards a feasible solution for the whole community.
+
+Technical leadership within a SIG should focus on solving the “How” questions
```
Comment (liggitt) [inline contributors/chairs-and-techleads/teachnical-lead.md:50]: nit: find/replace curly quotes everywhere:

```suggestion
Technical leadership within a SIG should focus on solving the "How" questions
```

**Review by liggitt — COMMENTED** [review]
lgtm, just a few nits

**Comment by dims** [comment]
@saschagrunert let's fix the last few nits and ship it. thanks for doing this!

File: contributors/chairs-and-techleads/teachnical-lead.md (hunk 7 above)
Comment (saschagrunert) [reply contributors/chairs-and-techleads/teachnical-lead.md]: Fixed by using the absolute URL :+1:

File: contributors/chairs-and-techleads/teachnical-lead.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
+# Technical Lead
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply contributors/chairs-and-techleads/teachnical-lead.md]: Ah, good catch. Fixed.

File: contributors/chairs-and-techleads/teachnical-lead.md (hunk 8 above)
Comment (saschagrunert) [reply contributors/chairs-and-techleads/teachnical-lead.md]: Changed as suggested.

**Comment by saschagrunert** [comment]
> @saschagrunert let's fix the last few nits and ship it. thanks for doing this!

Thanks for the hint, fixed the last review round as suggested.

**Comment by mrbobbytables** [comment]
/lgtm

As a follow-up (not needed in this PR), it might be worth linking the tech-lead description from [sig-governance.md](https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md#tech-lead) to this doc.

**Comment by dims** [comment]
/hold cancel

we have 3 steering members lgtm-ing :)

**Comment by justaugustus** [comment]
Great work, @saschagrunert!!

**Pull request merged**